src/os400qc3.h

```diff
@@ -360,16 +360,16 @@ extern int      _libssh2_bn_set_word(_libssh2_bn *bn, unsigned long val);
 extern int      _libssh2_bn_to_bin(_libssh2_bn *bn, unsigned char *val);
 extern int      _libssh2_random(unsigned char *buf, size_t len);
 extern void     _libssh2_os400qc3_crypto_dtor(_libssh2_os400qc3_crypto_ctx *x);
-extern int      libssh2_os400qc3_hash_init(Qc3_Format_ALGD0100_T *x,
-                                           unsigned int algo);
-extern void     libssh2_os400qc3_hash_update(Qc3_Format_ALGD0100_T *ctx,
-                                             const unsigned char *data,
-                                             int len);
-extern void     libssh2_os400qc3_hash_final(Qc3_Format_ALGD0100_T *ctx,
-                                            unsigned char *out);
-extern int      libssh2_os400qc3_hash(const unsigned char *message,
-                                      unsigned long len, unsigned char *out,
-                                      unsigned int algo);
+extern int      _libssh2_os400qc3_hash_init(Qc3_Format_ALGD0100_T *x,
+                                            unsigned int algo);
+extern void     _libssh2_os400qc3_hash_update(Qc3_Format_ALGD0100_T *ctx,
+                                              const unsigned char *data,
+                                              int len);
+extern void     _libssh2_os400qc3_hash_final(Qc3_Format_ALGD0100_T *ctx,
+                                             unsigned char *out);
```

Comment:
`_libssh2_os400qc3_hash_update` and `_libssh2_os400qc3_hash_final` now return an int.
After fixing these, compilation succeeds but I still have a `LIBSSH2_ERROR_KEY_EXCHANGE_FAILURE` at runtime: I have to check why in debugger.

Updating HACKING-CRYPTO according to this PR is needed too.

Reply:
The error is not caused by a hash function returnning 0: I have to check elsewhere... tomorrow!

Reply:
Thanks! I've now pushed fixes for these.

Something similar is also happening with libcrypto.